### DISCUSSION TITLE
问下在并发模型中连接协程到流协程分出来的是什么？

### DISCUSSION THREAD

**Discussion by zyong** [bfenetworks/bfe#837] [Q&A (问题和回答)] [answered]
你好，问下就是连接协程到流协程，发送过去的是具体单个流的处理逻辑还是什么别的？然后怎么控制连接的close呢？

**Comment by iyangsj** [comment, marked as answer]
1. 在HTTP2/SPDY连接中，每个流的请求及响应的处理，是在单独的流协程执行
参见：https://github.com/bfenetworks/bfe/blob/develop/bfe_http2/server.go#L1817 

2. 连接关闭时，连接协程在退出前将更新流的状态；相关的流协程在执行请求读取或响应发送操作，将出现IO错误并结束处理
参见：https://github.com/bfenetworks/bfe/blob/develop/bfe_spdy/server_conn.go#L790
